posit-dev/positron
# Data Explorer: Implement summary stats in Python for date and date time, add profile UI treatment for boolean, date, datetime

### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any GitHub issues that are related.

Addresses https://github.com/posit-dev/positron/issues/2161

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

Added comms definitions for date and datetime summary stats and implemented the python backend for computing
on pandas data frames.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues.

A minimal testing df:

```
import pandas as pd
df = pd.DataFrame({
    "x": [True, False, None],
    "y": pd.date_range("2000-01-01", periods=3).date,
    "z": pd.date_range("2000-01-01", periods=3),
    "i": pd.date_range("2000-01-01", periods=3, tz="US/Eastern")
})
%view df
```

Should be able to see summary stats:

![image](https://github.com/posit-dev/positron/assets/4706822/c866f2c1-5ab1-4c8d-a83a-7be5d1097f49)
